packages/ui-components/src/Button/Button.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FunctionComponent, ButtonHTMLAttributes } from "react";
+import classnames from "classnames/bind";
+
+import objectToClassnames from "../utils/objectToClassnames";
+
+import styles from "./Button.module.scss";
+
+export type ButtonType = "button" | "submit" | "reset";
+export type ButtonIntent =
+  | "primary-success"
+  | "primary-danger"
+  | "secondary"
+  | "tertiary";
+export type ButtonSize = "standard" | "small";
+
+type OwnButtonProps = {
+  // TODO(j-low): remove `type` prop and refactor tests accordingly
+  type?: ButtonType;
+  intent?: ButtonIntent;
+  size?: ButtonSize;
+  onClick?: () => void;
+};
+
+type NativeButtonProps = Omit<
+  ButtonHTMLAttributes<HTMLButtonElement>,
+  keyof OwnButtonProps
+>;
+
+export type ButtonProps = OwnButtonProps & NativeButtonProps;
+
+const cx = classnames.bind(styles);
+
+export const Button: FunctionComponent<ButtonProps> = ({
+  intent = "secondary",
+  type = "button",
```

Comment:
Since `type` is intersected from the native props, we can default the value without having it be a part of `OwnButtonProps`. But you **wil** need to include it in the button element ....